tgui/packages/tgui/interfaces/RuckingenurKit/index.tsx

```diff
@@ -0,0 +1,161 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Garash2k (https://github.com/Garash2k)
+ * @license MIT
+ */
+
+import { memo } from 'react';
+import { Button, Icon, Image, Section, Stack } from 'tgui-core/components';
+import { shallowDiffers } from 'tgui-core/react';
+
+import { useBackend } from '../../backend';
+import { Window } from '../../layouts';
+import {
+  BlueprintButtonStyle,
+  BlueprintMiniButtonStyle,
+} from '../Manufacturer/constant';
+import { RuckingenurKitData, ScannedItemData } from './type';
+
+export const RuckingenurKit = () => {
+  const { data } = useBackend<RuckingenurKitData>();
+  const { scanned_items, hide_allowed, olde } = data;
+
+  return (
+    <Window width={925} height={420}>
+      <Window.Content>
+        <Section title="Scanned Items" scrollable fill>
+          {scanned_items.map((scanned_item) => (
+            <ScannedItem
+              ScannedItem={scanned_item}
+              key={scanned_item.ref}
+              hide_allowed={hide_allowed}
+              olde={olde}
+            />
+          ))}
```

Comment:
Branch on (currently) `olde` here to render a different component instead if `olde` is truthy. That component is likely to be composed of a similar structure to `ScannedItem`, but you can then strip out any `olde`-specific stuff from `ScannedItem`.